### DISCUSSION TITLE
[2024-07-08] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#10475] [Weekly Demo Day]
* [#10169](https://github.com/enso-org/enso/issues/10169) - worked on electron part, where opening project in the same session was just not implemented. Then handled the dashboard part to @somebody1234 
* #10443

![image](https://github.com/enso-org/enso/assets/3919101/d8beb8d8-f828-482c-bd8b-0a8d86ae29d3)

* Almost merged: #10457 – we discard invalid input (it reverts to last valid state, so if someone type `0'` it will become `0`) - otherwise any input, even "fixed" might change irreversibly the widget structure.

**Comment by kazcw** [comment]
### Panels:
- #10385
- #10386

https://github.com/enso-org/enso/assets/1047859/19d64fed-74ab-48c2-af0e-48f140fa8474

### External documentation
- #10276

https://github.com/enso-org/enso/assets/1047859/24d7e8cc-3f5c-4644-9d11-7deb5d1a8767

### Resize handles
- #10292

https://github.com/enso-org/enso/assets/1047859/c9cb4cab-012b-4672-a277-16366aa4afc0

### Concurrent widgets fixes
- #10233 
- #10338 

### Little tweaks
- #10389
- #10232

**Comment by Frizi** [comment]
- Fixed issues with nested widget queries https://github.com/enso-org/enso/pull/10382
<img width="487" alt="image" src="https://github.com/enso-org/enso/assets/919491/af946c1c-a27f-4ed8-8346-b5098e7d5f08">

- Migrated all TypeScript workspace projects to use PNPM https://github.com/enso-org/enso/pull/10422
<img width="389" alt="image" src="https://github.com/enso-org/enso/assets/919491/649285cf-e4e6-49e5-9493-4df14dbedfca">

- Still in progress of fixing a bug with node ID reassignment causing node views to be recreated from scratch https://github.com/enso-org/enso/issues/10311

**Comment by hubertp** [comment]
Relevant backend changes: 
- Persist a subset of IdMap https://github.com/enso-org/enso/pull/10347
